mlrun/api/utils/builder.py

```diff
@@ -341,11 +342,13 @@ def build_image(
         if upgrade_pip_command:
             commands.append(upgrade_pip_command)
 
-        mlrun_command = resolve_mlrun_install_command(
+        mlrun_version = resolve_mlrun_install_command_version(
             mlrun_version_specifier, client_version, commands
         )
-        if mlrun_command:
-            commands.append(mlrun_command)
+
+        # mlrun must be installed with other python requirements in the same pip command to avoid version conflicts
+        if mlrun_version:
+            requirements_list = [mlrun_version] + requirements_list
```

Comment:
```suggestion
            requirements_list.insert(0, mlrun_version)
```

Reply:
funny, that's what I did at first but then I thought that the other way is more readable.
fixed